src/core/manifest.rs

```diff
@@ -39,14 +38,16 @@ impl Default for ComponentConfiguration {
 
 /// Representation of `manifest.json`
 #[allow(non_snake_case)]
-#[derive(Serialize, Deserialize, Clone, Default)]
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
 pub struct Manifest {
     /// Name of the main component
     pub name: String,
     /// Default environment to build in
     pub environment: String,
     /// All the environments dependencies can currently be found in
     pub supportedEnvironments: Vec<String>,
+    /// Custom environment definitions, or fallback to ~/.lal/config
+    pub environments: BTreeMap<String, Environment>,
```

Comment:
is there a main reason why you want this at the manifest level?
i feel like one of the biggest guarantees is the abi compatibility between targets that are built in the same env, will this not have the possibility of having equal keyed environments not match across manifests?